module/StraightRunner.cpp

```diff
@@ -11,27 +11,54 @@ StraightRunner::StraightRunner() {}
 // 設定された距離を直進する
 void StraightRunner::runStraightToDistance(double targetDistance, int pwm)
 {
-  double initialDistance = 0;
+  // 直進前の走行距離
+  int initialRightDistance = measurer.getRightCount();
+  int initialLeftDistance = measurer.getLeftCount();
+  double initialDistance = Mileage::calculateMileage(initialRightDistance, initialLeftDistance);
+  // 直進中の走行距離
+  int currentRightDistance = 0;
+  int currentLeftDistance = 0;
   double currentDistance = 0;
+  int error = 0;                // 左右の回転数の誤差
+  Pid pid(1.2, 0.3, 0.001, 0);  // 左右の回転数を合わせるためのPID
+  int adjustment = 0;           // 左右の誤差の補正値
 
-  // 初期値を格納
-  initialDistance = Mileage::calculateMileage(measurer.getRightCount(), measurer.getLeftCount());
+  // 現在のpwm値
+  int currentPwm = 0;
 
   // 距離の値が負または、pwm値が0の際は、終了する
   if(targetDistance < 0 || pwm == 0) return;
 
   // 走行距離が目標距離に到達するまで繰り返す
   while(true) {
     //現在の距離を取得する
-    currentDistance = Mileage::calculateMileage(measurer.getRightCount(), measurer.getLeftCount());
+    currentRightDistance = measurer.getRightCount();
+    currentLeftDistance = measurer.getLeftCount();
+    currentDistance = Mileage::calculateMileage(currentLeftDistance, currentRightDistance);
 
     //現在の距離が目標距離に到達したらループを終了する
     if(std::abs(currentDistance - initialDistance) >= targetDistance) {
       break;
     }
+
+    // PWM値を徐々に目標値に合わせる
+    if(currentPwm != pwm) {
+      // 調整距離毎にPWM値を加速値分だけ上げていく
+      currentPwm = (currentDistance + SECTION_DISTANCE) / SECTION_DISTANCE * ACCELE_PWM;
+      if(currentPwm > pwm) {
+        currentPwm = pwm;
+      }
+    }
+
+    // 左右の走行距離を合わせるた補正値計算
```

Comment:
修正する程のミスでもない気がしますが気になったので
```suggestion
    // 左右の走行距離を合わせるための補正値計算
```